src/index.cgi.go

```diff
@@ -0,0 +1,175 @@
+package main
+
+import (
+	"fmt"
+	"os"
+	"os/exec"
+	"net/http"
+	"net/http/cgi"
+	"encoding/json"
+	"bytes"
+	"log"
+	"strings"
+)
+
+func main() {
+	if err := cgi.Serve(http.HandlerFunc(cgiHandler)); err != nil {
+		log.Fatal(err)
+	}
+}
+
+func cgiHandler(w http.ResponseWriter, r *http.Request) {
+	if synoTokenRedirect(w, r) {
+		return
+	}
+
+	user, err := auth()
+	if err != nil {
+		http.Error(w, err.Error(), 403)
+		log.Fatal("tag 1: %v", err.Error())
+		return
+	}
+
+	s, err := tailscaleStatus()
+	if err != nil {
+		http.Error(w, err.Error(), 500)
+		log.Fatal("tag 2: %v", err.Error())
+		return
+	}
+	if uid := os.Getuid() ; uid != 0 {
```

Comment:
does this matter? I'm afraid in DSM7 they might start sandboxing these and actually running things as non-root. can we just drop this check?